File: dp/top k frequent elements.cpp
class Solution {
public:
    static bool cmp(pair<int, int>& a, pair<int, int>& b){
        return a.second > b.second;
    }
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int> mp;
        int size = nums.size();
        vector<pair<int,int>> v; 
        vector<int> ans;
        
        for(int i = 0; i < size; i++){
            mp[nums[i]]++;
        }
        
        for(auto& x:mp){
            v.push_back({x.first,x.second});
        }
        
        sort(v.begin(),v.end(),cmp); 
        
        for(int i=0;i<k;i++){
            ans.push_back(v[i].first);
        }
        return ans;
        
    }
};

File: leetcode/347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ans_map = {}

        for num in nums:
            if num in ans_map:
                ans_map[num] = ans_map[num] + 1
            else:
                ans_map[num] = 1

        s_map = sorted(ans_map.items(), key=lambda item: item[1], reverse=True)

        ans = []
        i = 0
        for key, value in s_map:
            if i < k:
                ans.append(key)
                i += 1

        return ans